lib/routes/sms.js

```diff
@@ -76,15 +76,17 @@ module.exports = (log, isA, error, config, customs, sms) => {
         function sendMessage (senderId) {
           return sms.send(phoneNumber, senderId, messageId, acceptLanguage)
             .catch(err => {
-              if (err.status === 500) {
-                throw error.messageRejected(err.reason, err.reasonCode)
+              if (err) {
+                if (err.status === 500) {
+                  throw error.messageRejected(err.reason, err.reasonCode)
+                }
+
+                if (err.status === 400) {
+                  throw error.invalidMessageId()
+                }
               }
 
-              if (err.status === 400) {
-                throw error.invalidMessageId()
-              }
-
-              throw new error.unexpectedError()
```

Comment:
Oh, probably because I didn't use the exact same function. It is indeed true that arrow functions cannot be used as constructors. I was using `unknownDevice`, which is a regular function. I'll grab my coat.